src/mongocxx/collection.cpp

```diff
@@ -1149,10 +1149,18 @@ cursor collection::_distinct(const client_session* session,
         throw bsoncxx::exception{bsoncxx::error_code::k_internal_error};
     }
 
+    const bson_t* error_document;
+
     cursor fake_cursor{libmongoc::cursor_new_from_command_reply_with_opts(
         _get_impl().client_impl->client_t, reply_bson, nullptr)};
-    if (libmongoc::cursor_error(fake_cursor._impl->cursor_t, &error)) {
-        throw_exception<operation_exception>(error);
+    if (libmongoc::cursor_error_document(fake_cursor._impl->cursor_t, &error, &error_document)) {
```

Comment:
For context, `cursor_error_document` is being called on a _fake_ cursor created for `distinct`. I don't _think_ it's even possible to have an error in the cursor here, as a legitimate error from sending `distinct` to the server would have surfaced a few blocks earlier...

Reply:
I agree that it isn't possible looking at the implementation of `mongoc_cursor_new_from_command_reply_with_opts`. Arguably it should set an error document reply if the reply was a server error, so I think this is still an improvment.